x-pack/plugins/ml/public/application/trained_models/nodes_overview/nodes_list.tsx

```diff
@@ -70,6 +66,14 @@ export const NodesList: FC = () => {
     try {
       const nodesResponse = await trainedModelsApiService.getTrainedModelsNodesOverview();
       setItems(nodesResponse.nodes);
+
+      // Update expanded rows.
+      nodesResponse.nodes.forEach((node) => {
+        if (itemIdToExpandedRowMap[node.id]) {
```

Comment:
Why are we mutating itemIdToExpandedRowMap here instead of using the setter?

Reply:
With a map, it doesn't matter much, because the reference of the object (i.e. jsx element) will be changed. It's also a bit faster